legato-api/src/main/java/org/opendaylight/legato/LegatoSlsProfileController.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2014 Cisco Systems, Inc. and others.  All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.opendaylight.legato;
+
+
+import java.util.Collections;
+
+import org.opendaylight.controller.md.sal.binding.api.DataBroker;
+import org.opendaylight.controller.md.sal.binding.api.DataTreeIdentifier;
+import org.opendaylight.controller.md.sal.binding.api.DataTreeModification;
+import org.opendaylight.controller.md.sal.common.api.data.LogicalDatastoreType;
+import org.opendaylight.legato.util.LegatoConstants;
+import org.opendaylight.legato.util.LegatoUtils;
+import org.opendaylight.unimgr.api.UnimgrDataTreeChangeListener;
+import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.MefGlobal;
+import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.SlsProfiles;
+import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.SlsProfilesBuilder;
+import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.sls.profiles.Profile;
+import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.global.rev171215.mef.global.sls.profiles.ProfileKey;
+import org.opendaylight.yangtools.concepts.ListenerRegistration;
+import org.opendaylight.yangtools.yang.binding.InstanceIdentifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+
+public class LegatoSlsProfileController  extends UnimgrDataTreeChangeListener<Profile> {
+	
+	public LegatoSlsProfileController(final DataBroker dataBroker) {
+		super(dataBroker);
+		registerListener();
+	}
+
+	private static final Logger LOG = LoggerFactory
+			.getLogger(LegatoSlsProfileController.class);
+	
+	
+	private static final InstanceIdentifier<Profile> PROFILE_IID = InstanceIdentifier
+			.builder(MefGlobal.class).child(SlsProfiles.class).child(Profile.class).build();
+	
+	
+	private static final InstanceIdentifier<SlsProfiles> SLS_PROFILES_IID_OPERATIONAL = InstanceIdentifier
+			.builder(MefGlobal.class).child(SlsProfiles.class).build();
+			
+	
+	private ListenerRegistration<LegatoSlsProfileController> dataTreeChangeListenerRegistration;
+	
+	
+	public void registerListener() {
+		LOG.info("Initializing LegatoSlsProfileController:init() ");
+
+		dataTreeChangeListenerRegistration = dataBroker.registerDataTreeChangeListener(
+				new DataTreeIdentifier<Profile>(LogicalDatastoreType.CONFIGURATION, PROFILE_IID), this);
+
+	}
+	
+
+	@Override
+	public void close() throws Exception {
+		if (dataTreeChangeListenerRegistration != null) {
+			dataTreeChangeListenerRegistration.close();
+		}
+		
+	}
+
+	
+	
+	@Override
+	public void add(DataTreeModification<Profile> newDataObject) {
+		if(newDataObject.getRootNode() != null && newDataObject.getRootPath() != null ){
```

Comment:
as getRootNode and getRootPath are marked @Notnull you do not need to check that. if you want to stay on the safe side you can use *assert* 
you have it elsewhere in the file. please fix it in the whole code base for legato module